pkg/replication/admission.go

```diff
@@ -489,15 +497,14 @@ func addAnnotation(key, value string) util.Patch {
 	}
 }
 
-func generateUpdated(operation admissionv1beta1.Operation, newSD *comp_v1.ServiceDescriptor) (time.Time, error) {
-	if operation != admissionv1beta1.Update {
+func generateUpdated(isReplication bool, operation admissionv1beta1.Operation, newSD *comp_v1.ServiceDescriptor) (time.Time, error) {
+	if operation != admissionv1beta1.Update || !isReplication {
```

Comment:
Hmm... So this code is written in a way that if replication controller doesn't set the timestamp (why? is that possible?), we will generate a timestamp.

Maybe instead we could always parseTimestamp for replication controller and return error if annotation is missing?

Reply:
afaict the only situation we *don't* want to generate a timestamp is if the replication controller is doing an update on an existing thing. In all other cases we do want to generate a timestamp. Isn't that what's going on here?

Reply:
Yes, but what is the case when replication controller doesn't provide a timestamp? I guess to rephrase the requirement: replication controller _must_ set timestamp and we don't touch it (but validate it's set). In all other cases we generate timestamp.
This way it will be easier to catch bugs in replication controller I think

Reply:
Fair call, I'll fix that up

Reply:
Also not sure if we actually still need to check for `operation != admissionv1beta1.Update`? I find this code hard to understand. My idea:
```
if isReplication {
  ts, exists := newSD.Annotations[updatedKey]
  if !exists {
    return nil, errors.New("replication controller must set the mutation timestamp")  
  }
  return parseTimestamp(ts)
}

return time.Now(), nil
```